### PR TITLE
#3193[Create events]/Add validations and dashboard

### PR DESCRIPTION
**Pull request by Mukola1983**
[#4236](https://github.com/ita-social-projects/GreenCity/issues/4236#issue-1250584127) [Event dashboard]/User shall be able to view all the events
[#4234](https://github.com/ita-social-projects/GreenCity/issues/4234#issue-1250568490) Add styles to create events
[#4232](https://github.com/ita-social-projects/GreenCity/issues/4232#issue-1250557225) Add locations and online link and validation
[#4231](https://github.com/ita-social-projects/GreenCity/issues/4231#issue-1250543196) Validation event title, date and time fields 
[#4235](https://github.com/ita-social-projects/GreenCity/issues/4235#issue-1250571644) Unit tests
[#4228](https://github.com/ita-social-projects/GreenCity/issues/4228#issue-1250507486) Authorized User shall be able to create an event
[#4230](https://github.com/ita-social-projects/GreenCity/issues/4230#issue-1250528592) User should be able upload image
[#4229](https://github.com/ita-social-projects/GreenCity/issues/4229#issue-1250516503) User should fill title, date,time,description

### CODE REVIEW

**Review by NewArtMask — APPROVED** [review]

File: src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts [hunk 1]
```diff
@@ -31,82 +35,131 @@ export class CreateEditEventsComponent implements OnInit {
...
+  public changeToClose(): void {
     this.isOpen = false;
   }
 
-  setDateCount(ev: MatSelectChange): void {
-    this.dates.length = +ev.value.split(' ')[0];
+  public setDateCount(event: MatSelectChange): void {
+    this.dates.length = +event.value.split(' ')[0];
 
     for (let i = 0; i < this.dates.length; i++) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts:96]: use this.dates.map here

File: src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts [hunk 2]
```diff
@@ -31,82 +35,131 @@ export class CreateEditEventsComponent implements OnInit {
...
 
-  public changedEditor(event: EditorChangeContent | EditorChangeSelection): void {
-    if (event.event !== 'selection-change') {
-      this.editorHTML = event.html;
+  private checkDates() {
+    this.dates.forEach((item) => {
+      item.check = !item.valid;
+    });
+
+    for (const value of this.dates) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts:133]: remove for..of cycle here, use some method for dates array
this.checkdates = !this.dates.some((item) => !item.valid)

File: src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts [hunk 3]
```diff
@@ -115,21 +168,51 @@ export class CreateEditEventsComponent implements OnInit {
...
-    const stringifiedDataToSend = JSON.stringify(sendEventDto);
-    formData.append('addEventDtoRequest', stringifiedDataToSend);
-    for (const images of this.imgArray) {
-      formData.append('images', images);
+    if (this.checkdates && this.titleForm.valid && this.contentValid) {
+      this.checkAfterSend = true;
+      const formData: FormData = new FormData();
+      const stringifiedDataToSend = JSON.stringify(sendEventDto);
+      formData.append('addEventDtoRequest', stringifiedDataToSend);
+      for (const images of this.imgArray) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts:196]: use forEach here

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 4]
```diff
@@ -1,20 +1,48 @@
...
       this.eventsList = [...res.page];
+      this.total = res.totalElements;
     });
   }
+
+  private checkUserSingIn(): void {
+    this.userOwnAuthService.credentialDataSubject.subscribe((data) => (this.isLoggedIn = data && data.userId));
+  }
+
+  public setPage(event: number) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts:37]: add return type (void)

File: src/app/main/component/events/events.module.ts [hunk 5]
```diff
@@ -38,12 +40,15 @@ import { GooglePlaceModule } from 'ngx-google-places-autocomplete';
     ImagesContainerComponent
   ],
   imports: [
+    ReactiveFormsModule,
+
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/events.module.ts:44]: remove empty row

**Review by hnativlyubomyr — APPROVED** [review]

**Review by voronkooleksandr — APPROVED** [review]

**Review by krayevska — APPROVED** [review]

**Review by Myh45 — APPROVED** [review]

**Pull request merged**